sdk/textanalytics/Azure.AI.TextAnalytics/src/Transforms.cs

```diff
@@ -145,9 +159,23 @@ internal static ExtractKeyPhrasesResultCollection ConvertToExtractKeyPhrasesResu
             }
 
             //Read Key phrases
-            foreach (KeyPhraseResultDocumentsItem docKeyPhrases in results.Documents)
-            {
-                keyPhrases.Add(new ExtractKeyPhrasesResult(docKeyPhrases.Id, docKeyPhrases.Statistics ?? default, ConvertToKeyPhraseCollection(docKeyPhrases)));
+            foreach (KeyPhraseResultDocumentsItem document in results.Documents)
+            {
+                DetectedLanguage? detectedLanguage = (document.DetectedLanguage is not null)
+                    ? new DetectedLanguage(
+                        document.DetectedLanguage.Value.Name,
+                        document.DetectedLanguage.Value.Iso6391Name,
+                        document.DetectedLanguage.Value.ConfidenceScore,
+                        document.DetectedLanguage.Value.Script,
```

Comment:
just curious, if someone sets the api version to an older version in which script is not supported/sent by the service, will this code set it to null?

Reply:
My understanding is that both automatic language detection and script detection are both new features in version 2022-10-01-preview, which means that if a version older than 2022-10-01-preview is used, the request would fail if "auto" is passed in the request (or at a minimum, detectedLanguage would be null in the response). Is this correct?

Reply:
I haven't tested "auto" in an older version yet so not sure what the service behavior is. In python, if an older version is selected, the generated model received by the transform code won't have a detectedLanguage property and so we set it to null/None on the exposed model. I guess I'm wondering in .NET's case if you would still create the DetectedLanguage model, but just set each of its properties to null.